airflow/hooks/oracle_hook.py

```diff
@@ -0,0 +1,79 @@
+import cx_Oracle
+import logging
+
+from airflow.hooks.dbapi_hook import DbApiHook
+
+class OracleHook(DbApiHook):
+    """
+    Interact with Oracle SQL Server.
+    """
+    conn_name_attr = 'oracle_conn_id'
+    default_conn_name = 'oracle_default'
+    supports_autocommit = False
+
+    def get_conn(self):
+        """
+        Returns a oracle connection object
+        """
+        conn = self.get_connection(self.oracle_conn_id)
+        try:
+            conn = cx_Oracle.connect(conn.login, conn.password, conn.host)
+        except:
+            # add support for custom DSN connections
+            dsn = conn.extra_dejson.get('dsn', None)
+            service_name = conn.extra_dejson.get('service_name', None)
+            dsn = cx_Oracle.makedsn(dsn, conn.port, service_name=service_name)
+            conn = cx_Oracle.connect(conn.login, conn.password, dsn=dsn)
+        return conn
+
+    def get_pandas_df(self, sql, parameters = None):
```

Comment:
I'm not sure if the lower case change is necessary, but if you think it is you should call `super` and then alter the column list to avoid duplicating the rest of the code.